library/network_connections.py

```diff
@@ -948,6 +961,14 @@ def connection_create(self, connections, idx, connection_current=None):
                         s_ethtool.option_set(nm_coalesce, None)
                     else:
                         s_ethtool.option_set_uint32(nm_coalesce, int(setting))
+            for ring, setting in connection["ethtool"]["ring"].items():
+                nm_ring = nm_provider.get_nm_ethtool_ring(ring)
+
+                if nm_ring:
+                    if setting is None:
+                        s_ethtool.option_set(nm_ring, None)
+                    else:
+                        s_ethtool.option_set_uint32(nm_ring, setting)
```

Comment:
yes, right. But as I tried to say earlier, above is also
```python
         for coalesce, setting in ethtool_coalesce.items():
             if setting is not None:
                 if isinstance(setting, bool):
                     setting = int(setting)
                 configured_coalesce.append(
```
with the same problem. Please fix this pre-existing code in a separate commit.

Reply:
Great observation @liangwen12year - there might be a problem with initscripts support for coalesce, though, since I guess it needs on or off for boolean values instead of 1/0 (not sure).

Reply:
Please open a new issue or PR for tracking the coalesce efforts instead of this PR.

Reply:
New issue created, https://github.com/linux-system-roles/network/issues/388